### PR TITLE
add more stack size features + countdowncalculator support for crystal hollows pass tooltip

### PR DESCRIPTION
**Pull request by RayDeeUx**

### CODE REVIEW

**Comment by RayDeeUx** [comment]
pull-request-quantifier you have no bitches too shut up

File: src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/ItemFeatures.kt [hunk 1]
```diff
@@ -590,6 +590,17 @@ class ItemFeatures {
                 }
             }
         }
+        if (Skytils.config.showCollectionStackSize) {
+            val collectionDisplayName = getDisplayName(item)
+            val collectionMatcher = collectionPattern.matcher(collectionDisplayName)
```
Comment (Sychic) [inline src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/ItemFeatures.kt:596]: Where is collectionPattern?

File: src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/ItemFeatures.kt [hunk 2]
```diff
@@ -590,6 +590,17 @@ class ItemFeatures {
...
+            if (collectionMatcher.matches()) {
+                lore?.forEach { line -> //prevent false positives
+                    if (line.contains("Collection")) {
+                        stackTip = collectionMatcher.group(2).toString().romanToDecimal().toString()
+                    }
+                }
+            }
```
Comment (Sychic) [inline src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/ItemFeatures.kt]: Purely guessing that there are 3 groups
```suggestion
            collectionPattern.matchEntire(getDisplayName(item))?.destructured?.let { (_, _, level) ->
                lore?.forEach { line ->
                    if (line.contains("Collection")) {
                        stackTip = level.romanToDecimal().toString()
                        return@let
                    }
                }
            }
```

File: src/main/kotlin/gg/skytils/skytilsmod/utils/ItemUtil.kt [hunk 3]
```diff
@@ -29,6 +29,7 @@ import kotlin.math.max
 
 object ItemUtil {
     private val PET_PATTERN = "§7\\[Lvl \\d+] (?<color>§[0-9a-fk-or]).+".toRegex()
+    private val PET_PATTERN_FOR_SYNTHESIS = "§7\\[\\d+] (?<color>§[0-9a-fk-or]).+".toRegex()
```
Comment (Sychic) [inline src/main/kotlin/gg/skytils/skytilsmod/utils/ItemUtil.kt]: We won't be adding support for non-hypixel values.

**Review by Sychic — CHANGES_REQUESTED** [review]

File: src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/ItemFeatures.kt (hunk 1 above)
Comment (RayDeeUx) [reply src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/ItemFeatures.kt:595]: GAH! forgot that one
`private val collectionPattern = Pattern.compile(".*(?<collection1>[A-Z][a-z]+) (?<tier>[X,I,V]+)")`

**Comment by RayDeeUx** [comment]
@Sychic check again, things should be fine now.

also my edits to countdowncalculator.kt are mostly cosmetic (comment blocks or string changes) for readability + instruction manual for future folks

File: src/main/kotlin/gg/skytils/skytilsmod/utils/ItemUtil.kt (hunk 3 above)
Comment (RayDeeUx) [reply src/main/kotlin/gg/skytils/skytilsmod/utils/ItemUtil.kt:32]: got it.

File: src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/ItemFeatures.kt (hunk 2 above)
Comment (RayDeeUx) [reply src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/ItemFeatures.kt]: nope, two.

**Pull request closed without merging**